File: smartmedika.admin/src/app/model/paginationEntity.ts
export class PaginationEntity<T> {
	public page: number = 0;
	public totalPages: number = 0;
	public totalCount: number = 0;
	public count: number = 0;
	public pageSize: number = 0;
    public items: Array<T>;

	constructor(page: number = 0, pageSize:number = 0, totalPages: number = 0, totalCount: number = 0, count:number = 0) {
		this.page = page;
		this.pageSize = pageSize;
		this.totalPages = totalPages;
		this.totalCount = totalCount;
        this.count = count;
	}	

	get offset():number{
		let offset =  (this.page - 1) * this.pageSize;
		return offset;
	}
}

File: server/dnacorebase/Dna.Core.Base.Data/PaginationEntity.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Dna.Core.Base.Data
{
    public class PaginationEntity<T>
    {
        public int Page { get; set; }
        public int PageSize { get; set; }

        public int Count
        {
            get
            {
                return (null != this.Items) ? this.Items.Count() : 0;
            }
        }


        public int TotalCount { get; set; }
        public int TotalPages
        {
            get
            {
                return (int)Math.Ceiling((decimal)TotalCount / PageSize);
            }
        }

        public IEnumerable<T> Items { get; set; }
    }
}
